client/state/comments/selectors.js

```diff
@@ -47,8 +43,9 @@ export const getPostTotalCommentsCount = ( state, siteId, postId ) => state.comm
  */
 export const getPostMostRecentCommentDate = createSelector(
 	( state, siteId, postId ) => {
-		const items = getPostCommentItems( state, siteId, postId );
-		return items && items.first() ? new Date( items.first().get( 'date' ) ) : undefined;
+		const items = filter( getPostCommentItems( state, siteId, postId ), { parent: false } );
```

Comment:
also, if the comment items are still an array of all comments sorted by date, you don't need this walk at all. You can just take the first item and use that, since the array is sorted. And vice-versa for get oldest date.